packages/sdk-js/src/GrowthBook.ts

```diff
@@ -259,24 +265,56 @@ export class GrowthBook<
     this.setExperiments(JSON.parse(experimentsJSON) as AutoExperiment[]);
   }
 
+  public async decryptPayload(
+    data: FeatureApiResponse,
+    decryptionKey?: string,
+    subtle?: SubtleCrypto
+  ): Promise<FeatureApiResponse> {
+    if (data.encryptedFeatures) {
+      data.features = JSON.parse(
+        await decrypt(
+          data.encryptedFeatures,
+          decryptionKey || this._ctx.decryptionKey,
+          subtle
+        )
+      );
+      delete data.encryptedFeatures;
+    }
+    if (data.encryptedExperiments) {
+      data.experiments = JSON.parse(
+        await decrypt(
+          data.encryptedExperiments,
+          decryptionKey || this._ctx.decryptionKey,
+          subtle
+        )
+      );
+      delete data.encryptedExperiments;
+    }
+    return data;
+  }
+
   public setAttributes(attributes: Attributes) {
     this._ctx.attributes = attributes;
-    if (this._ctx.remoteEval) {
-      this._refreshForRemoteEval();
-      return;
-    }
-    this._render();
-    this._updateAllAutoExperiments();
+    this.refreshStickyBuckets().then(() => {
```

Comment:
risky? setAttributes is no longer synchronous, even if sticky bucketing is disabled (side effects happen via `.then()` in the next loop cycle).

Reply:
actually, any qualms with making this function async (since it's basically 90% async already without returning its promise)?